swap/swap.go

```diff
@@ -252,10 +272,24 @@ func (swap *Swap) GetMetricTypes(cfg plugin.ConfigType) ([]plugin.MetricType, er
 		metricType := plugin.MetricType{Namespace_: core.NewNamespace(vendorPrefix, srcPrefix, typePrefix, ioPrefix, metric)}
 		metricTypes = append(metricTypes, metricType)
 	}
+	// List of terminal metric names
+	mList := make(map[string]bool)
 	for _, device := range devices {
```

Comment:
@thomastaylor312 I think you are right as furthermore the probability of devices returning different sets of metrics should be very close (if not equal) to 0
In fact, I did not want to change the code too much therefore I kept the loop...
Will simplify this too in forthcomming submission

Note that unfortunately some of my other PRs also contained similar code :-(
[see this one for instance
](https://github.com/intelsdi-x/snap-plugin-collector-interface/pull/12/commits/14a3e1f249b4b7f6bdd78a9c00ce563b4e721f7e)